File: solution/ServerExam/Entity/PaperRuleEntity.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace ServerExam.Entity
{
    public class PaperRuleEntity
    {
        [Key]
        [JsonPropertyName("ID")]
        public int ID { get; set; }

        /// <summary>
        /// 大标题ID
        /// </summary>
        [JsonPropertyName("HeadlineID")]
        public int HeadlineID { get; set; }

        /// <summary>
        /// 试题类型
        /// </summary>
        [JsonPropertyName("QuestionType")]
        public int QuestionType { get; set; }

        /// <summary>
        /// 试题数量
        /// </summary>
        [JsonPropertyName("QuestionNum")]
        public int QuestionNum { get; set; }

        /// <summary>
        /// 平均分
        /// </summary>
        [JsonPropertyName("AverageScore")]
        public decimal AverageScore { get; set; }

        /// <summary>
        /// 试卷ID
        /// </summary>
        [JsonPropertyName("PaperID")]
        public int PaperID { get; set; }

        /// <summary>
        /// 试卷规则状态 1正常 2禁用
        /// </summary>
        [JsonPropertyName("PaperRuleState")]
        public int PaperRuleState { get; set; }

        /// <summary>
        /// 创建时间
        /// </summary>
        [JsonPropertyName("CreateTime")]
        public int CreateTime { get; set; }

        /// <summary>
        /// 更新时间
        /// </summary>
        [JsonPropertyName("UpdateTime")]
        public int UpdateTime { get; set; }

        public PaperRuleEntity()
        {
            this.ID = 0;
            this.HeadlineID = 0;
            this.QuestionType = 0;
            this.QuestionNum = 0;
            this.AverageScore = 0;
            this.PaperID = 0;
            this.CreateTime = 0;
            this.UpdateTime = 0;
        }
    }
}

File: solution/Entity/PaperRuleEntity.py
from Entity.BaseEntity import *


# 试卷规则
class PaperRuleEntity(BaseEntity, BaseORM):
    __tablename__ = 'PaperRule'

    HeadlineID: int = Column(INTEGER, comment='大标题ID', default=0)
    QuestionType: int = Column(INTEGER, comment='试题类型 1单选 2判断 3多选 4填空 5问答', default=0)
    QuestionNum: int = Column(INTEGER, comment='抽题数量', default=0)
    SingleScore: float = Column(DECIMAL(10, 2), comment='单题分数', default=0.00)
    PaperID: int = Column(INTEGER, comment='试卷ID', default=0)
    PaperRuleState: int = Column(INTEGER, comment='试卷规则状态 1正常 2禁用', default=0)
    UpdateTime: int = Column(INTEGER, comment='更新时间', default=int(time()))

    def __init__(self):
        super().__init__()